ext/afform/core/CRM/Afform/Page/AfformBase.php

```diff
@@ -9,11 +9,16 @@ public function run() {
     // To avoid php complaints about the number of args passed to this function vs the base function
     [$pagePath, $pageArgs] = func_get_args();
 
-    // The api will throw an exception if afform is not found (because of the index 0 param)
+    $formName = $pageArgs['afform'] ?? NULL;
+    if (!$formName && $this->urlPath[0] === 'civicrm' && $this->urlPath[1] === 'afform' && $this->urlPath[2] === 'view') {
+      $formName = $this->urlPath[3];
+    }
```

Comment:
@totten the other half of the heart of the PR is here, where the afform base page is routed both from the form-specific server_routes, and also from the generic route. This patch here handles the generic route.

FYI to clarify your question about the navigation menu and conditionality, I think you may have misread that bit. No it's not conditional. The generic route is available for every form.
Effectively that means that if you give your form a server_route, it is not availble in two places.
This is the same as for e.g. drupal nodes, where if you give it a vanity url it is now available a `/my-cool-page` and it is *also* still available at `/node/123`.
The rest of the PR is just updating the GUI to account for the availability of the generic routes.